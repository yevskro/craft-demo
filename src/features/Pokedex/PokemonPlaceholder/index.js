import React from 'react';
import styled from 'styled-components';

function PokemonPlaceholder() {
  return (
    <StyledPokemonPlaceholder>
      <ImagePlaceholder />
      <NamePlaceholder />
    </StyledPokemonPlaceholder>
  );
}

const StyledPokemonPlaceholder = styled.div`
  width: 96px;
  display: inline-block;
`;

const ImagePlaceholder = styled.div`
  box-sizing: border-box;
  border: 15px solid white;
  width: 96px;
  height: 96px;
  background-color: rgba(233, 233, 233, 0.8);
`;

const NamePlaceholder = styled.div`
  box-sizing: border-box;
  background-color: rgba(233, 233, 233, 0.8);
  height: 19px;
  width: 96px;
  border: 5px solid white;
`;
export default PokemonPlaceholder;
