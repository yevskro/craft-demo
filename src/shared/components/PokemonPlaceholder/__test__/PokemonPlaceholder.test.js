import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PokemonPlaceholder from '..';

test('renders without crashing', () => {
  render(<PokemonPlaceholder />);
});

test('snapshot', async () => {
  const tree = renderer.create(<PokemonPlaceholder />).toJSON();
  expect(tree).toMatchSnapshot();
});
