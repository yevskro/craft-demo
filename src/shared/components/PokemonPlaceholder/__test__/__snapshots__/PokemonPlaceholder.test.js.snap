// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  width: 96px;
  display: inline-block;
}

.c1 {
  box-sizing: border-box;
  border: 15px solid white;
  width: 96px;
  height: 96px;
  background-color: rgba(233,233,233,0.8);
}

.c2 {
  box-sizing: border-box;
  background-color: rgba(233,233,233,0.8);
  height: 19px;
  width: 96px;
  border: 5px solid white;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
</div>
`;
